ksql-functional-tests/src/test/resources/query-validation-tests/udf-tests.json

```diff
@@ -0,0 +1,119 @@
+        {"topic": "OUTPUT", "value": {"R0" : -1, "R00" : -1.0, "R1" :  -1.5, "R2" : -1.5,  "R10" :  -1.5, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -2, "R00" : -2.0, "R1" :  -1.7, "R2" : -1.7,  "R10" :  -1.7, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 12346, "R00" : 12346.0, "R1" :  12345.7, "R2" : 12345.68,  "R10" :  12345.678, "1R" :  12350.0 ,"2R" :  12300.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -12346, "R00" : -12346.0, "R1" :  -12345.7, "R2" : -12345.68,  "R10" :  -12345.678, "1R" :  -12350.0 ,"2R" :  -12300.0 }}
+      ]
+    },
+    {
+      "name": "calculate ROUND with large DECIMAL values",
+      "statements": [
+        "CREATE STREAM test (v DECIMAL(33, 16)) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE STREAM OUTPUT AS SELECT ROUND(v) as R0, ROUND(v, 0) as R00, ROUND(v, 1) as R1, ROUND(v, 2) as R2, ROUND(v, 10) as R10, ROUND(v , -1) as 1R , ROUND(v , -2) as 2R  FROM test;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "value": "12345678987654321.2345678987654321"}
+      ],
+      "outputs": [
+        {
+          "topic": "OUTPUT",
+          "value": "\"12,345,678,987,654,321\",\"12,345,678,987,654,321.0000000000000000\",\"12,345,678,987,654,321.2000000000000000\",\"12,345,678,987,654,321.2300000000000000\",\"12,345,678,987,654,321.2345678988000000\",\"12,345,678,987,654,320.0000000000000000\",\"12,345,678,987,654,300.0000000000000000\""
+        }
+      ],
+      "post": {
+        "sources": [
+          {
+            "name": "OUTPUT",
+            "type": "stream",
+            "valueSchema": "STRUCT<R0 DECIMAL(17,0), R00 DECIMAL(33,16), R1 DECIMAL(33,16), R2 DECIMAL(33,16), R10 DECIMAL(33,16), 1R DECIMAL(33,16), 2R DECIMAL(33,16)>"
+          }
+        ]
+      }
+    },
+    {
+      "name": "calculate FLOOR function",
+      "statements": [
+        "CREATE STREAM test (v DOUBLE) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT FLOOR(v) as F0 FROM test;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "value": {"v" : 1.2}},
+        {"topic": "test_topic", "value": {"v" : 1.7}},
+        {"topic": "test_topic", "value": {"v" : 1.5}},
+        {"topic": "test_topic", "value": {"v" : 3}},
+        {"topic": "test_topic", "value": {"v" : 1.234567}},
+        {"topic": "test_topic", "value": {"v" : 0.0}},
+        {"topic": "test_topic", "value": {"v" : null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": {"F0" : 1.0}},
+        {"topic": "OUTPUT", "value": {"F0" : 1.0}},
+        {"topic": "OUTPUT", "value": {"F0" : 1.0}},
+        {"topic": "OUTPUT", "value": {"F0" : 3.0}},
+        {"topic": "OUTPUT", "value": {"F0" : 1.0}},
+        {"topic": "OUTPUT", "value": {"F0" : 0.0}},
+        {"topic": "OUTPUT", "value": {"F0" : null}}
+      ]
+    },
+    {
+      "name": "calculate CEIL function",
```

Comment:
As per test for floor in `math.json`, it would be nice if this test covered calling `CEIL` with `INT`, `BIGINT` and `DECIMAL` as well as `DOUBLE`.

Where testing `DECIMAL` I'd add a quick `post` block to confirm schemas, as before.

Reply:
Tbh, I don't really know why there is a test for CEIL in this PR as the changes do not involve any changes to CEIL but it was provided in the users PR.
It also seems that the current CEIL implementation only works with double anyway.